pkg/kubelet/container/runtime.go

```diff
@@ -46,12 +46,10 @@ type Runtime interface {
 	GetPods(all bool) ([]*Pod, error)
 	// RunPod starts all the containers of a pod within a namespace.
 	RunPod(*api.Pod, map[string]volume.Volume) error
+	// Syncs the running pod into the desired pod.
+	SyncPod(pod *api.Pod, runningPod Pod, podStatus api.PodStatus) error
```

Comment:
@yifan-gu, why is it more efficient to pass `runningPod` to `SyncPod`? Were you talking about the runtime cache? I am skeptical that we need to have the runtime cache in pod_workers.go, as opposed to in the separate container runtime. `GetPods` is relatively more expensive for docker, but that doesn't mean it's the same for rkt or other container runtimes. It seems that it makes more sense for each container runtime to optimize their operations individually.

I agree that it might be less disruptive to keep `runningPod` for now though. @vmarmol, what's your opinion on this?